### PR TITLE
feat(spoolman): QR code scanning support

### PR DESCRIPTION
**Pull request by matmen**
As described in https://github.com/fluidd-core/fluidd/pull/1119

Adds support for spoolman QR codes (https://github.com/Donkie/Spoolman/issues/10).
Adds an option to automatically commit a spool selection when a valid QR code is scanned (no need to manually confirm the selection)
Adds support for using the device's camera as a scan source (https://github.com/fluidd-core/fluidd/pull/1119#issuecomment-1676159359)
Closes https://github.com/fluidd-core/fluidd/issues/1151

### CODE REVIEW

File: src/components/widgets/spoolman/QRReader.vue [hunk 1]
```diff
@@ -34,7 +34,11 @@ import BrowserMixin from '@/mixins/browser'
   components: { CameraItem }
 })
 export default class QRReader extends Mixins(StateMixin, BrowserMixin) {
-  dataPatterns = [/\/spool\/show\/(\d+)\/?/]
+  dataPatterns = [
+    /web\+spoomnan:s-(\d+)/,
```
Comment (pedrolamas) [inline src/components/widgets/spoolman/QRReader.vue:38]:
```suggestion
    /web\+spoolman:s-(\d+)/,
```

**Review by pedrolamas — COMMENTED** [review]
One typo, once that is fixed this is good to go!

**Review by pedrolamas — APPROVED** [review]

**Comment by matmen** [comment]
@NonaSuomy Feel free to test these changes, I just added support for using the device's camera as a scan source

**Comment by pedrolamas** [comment]
@matmen To fix #1150, I made `crossorigin` an optional property, so you will need to add `crossorigin="anonymous"` to [this element](https://github.com/fluidd-core/fluidd/pull/1149/files#diff-cd6975ab9d4451c59e4630210657f1df8a4f930f6a901edd948f187b20d06490R16)

Having said that we now know that this will still fail if the streaming server is not setting the `Access-Control-Allow-Origin "*"` response header... in that case the DOMException that you are already catching will have `ex.name === "SecurityError"`

**Comment by pedrolamas** [comment]
And while we're at it, we might want to add #1151 here?

**Review by pedrolamas — APPROVED** [review]

**Pull request merged**